src/find_zero.jl

```diff
@@ -271,90 +271,84 @@ end
+function (F::Callable_Function{Val{3},Val{true},𝑭,Nothing})(x) where {𝑭}
     f, f′, f′′ = (F.f[1])(x), (F.f[2])(x), (F.f[3])(x)
     f, (f/f′, f′/f′′)
 end
-(F::Callable_Function{S,T,𝑭,P})(x) where {S <: Val{3}, T <: Val{true}, 𝑭, P} = begin
+function (F::Callable_Function{Val{3},Val{true},𝑭,P})(x) where {𝑭, P}
     f, f′, f′′ = (F.f[1])(x, F.p), (F.f[2])(x, F.p), (F.f[3])(x, F.p)
     f, (f/f′, f′/f′′)
 end
 
-(F::Callable_Function{S,T,𝑭,P})(x) where {S <: Val{4}, T <: Val{true}, 𝑭, P<:Nothing} = begin
+function (F::Callable_Function{Val{4},Val{true},𝑭,Nothing})(x) where {𝑭}
     f, f′, f′′, f′′′ = (F.f[1])(x), (F.f[2])(x), (F.f[3])(x), (F.f[4])(x)
     f, (f/f′, f′/f′′, f′′/f′′′)
 end
-(F::Callable_Function{S,T,𝑭,P})(x) where {S <: Val{4}, T <: Val{true}, 𝑭, P} = begin
+function (F::Callable_Function{Val{4},Val{true},𝑭,P})(x) where {𝑭, P}
     f, f′,f′′,f′′′ = (F.f[1])(x, F.p), (F.f[2])(x, F.p), (F.f[3])(x, F.p), (F.f[4])(x, F.p)
     𝐓 = eltype(f/f′)
     f, NTuple{3,𝐓}((f/f′, f′/f′′, f′′/f′′′))
 end
 
-(F::Callable_Function{S,T,𝑭,P})(x) where {S <: Val{5}, T <: Val{true}, 𝑭, P<:Nothing} = begin
+function (F::Callable_Function{Val{5},Val{true},𝑭,Nothing})(x) where {𝑭}
     f, f′, f′′, f′′′, f′′′′ = (F.f[1])(x), (F.f[2])(x), (F.f[3])(x), (F.f[4])(x), (F.f[5])(x)
     f, (f/f′, f′/f′′, f′′/f′′′, f′′′/f′′′′)
 end
-(F::Callable_Function{S,T,𝑭,P})(x) where {S <: Val{5}, T <: Val{true}, 𝑭, P} = begin
+function (F::Callable_Function{Val{5},Val{true},𝑭,P})(x) where {𝑭, P}
     f, f′,f′′,f′′′,f′′′′ = (F.f[1])(x, F.p), (F.f[2])(x, F.p), (F.f[3])(x, F.p), (F.f[4])(x, F.p), (F.f[5])(x, F.p)
     f, (f/f′, f′/f′′, f′′/f′′′, f′′′/f′′′′)
 end
 
-(F::Callable_Function{S,T,𝑭,P})(x) where {S <: Val{6}, T <: Val{true}, 𝑭, P<:Nothing} = begin
+function (F::Callable_Function{Val{6},Val{true},𝑭,Nothing})(x) where {𝑭}
     f, f′, f′′, f′′′, f′′′′, f′′′′′ = (F.f[1])(x), (F.f[2])(x), (F.f[3])(x), (F.f[4])(x), (F.f[5])(x),(F.f[6])(x)
     f, (f/f′, f′/f′′, f′′/f′′′, f′′′/f′′′′, f′′′′/f′′′′′)
 end
-(F::Callable_Function{S,T,𝑭,P})(x) where {S <: Val{6}, T <: Val{true}, 𝑭, P} = begin
+function (F::Callable_Function{Val{6},Val{true},𝑭,P})(x) where {𝑭, P}
     f, f′,f′′,f′′′,f′′′′, f′′′′′ = (F.f[1])(x, F.p), (F.f[2])(x, F.p), (F.f[3])(x, F.p), (F.f[4])(x, F.p), (F.f[5])(x, F.p), (F.f[6])(x, F.p)
     f, (f/f′, f′/f′′, f′′/f′′′, f′′′/f′′′′, f′′′′/f′′′′′)
 end
 
 # faster with the above written out, should generate them...
-(F::Callable_Function{S,T,𝑭,P})(x) where {𝐍, S <: Val{𝐍}, T <: Val{true}, 𝑭, P<:Nothing} = begin
+function (F::Callable_Function{Val{𝐍},Val{true},𝑭,Nothing})(x) where {𝐍, 𝑭}
     fs = ntuple(i -> F.f[i](x), Val(𝐍))
     first(fs), ntuple(i -> fs[i]/fs[i+1], Val(𝐍-1))
 end
 
-(F::Callable_Function{S,T,𝑭,P})(x) where {𝐍, S <: Val{𝐍}, T <: Val{true}, 𝑭, P} = begin
-    fs = ntuple(i -> F.f[i](x, p), Val(𝐍))
+function (F::Callable_Function{Val{𝐍},Val{true},𝑭,P})(x) where {𝐍, 𝑭, P}
+    fs = ntuple(i -> F.f[i](x, F.p), Val(𝐍))
```

Comment:
This was a bug that was revealed by the language server.

Reply:
Wow, better living through better tools...